hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/tracing/TraceUtils.java

```diff
@@ -72,4 +97,71 @@ private String getInternal(String key) {
       }
     };
   }
+
+  public static Tracer createAndRegisterTracer(String name) {
+    if (!GlobalTracer.isRegistered()) {
+      io.jaegertracing.Configuration config =
+          io.jaegertracing.Configuration.fromEnv(name);
+      Tracer tracer = config.getTracerBuilder().build();
+      GlobalTracer.register(tracer);
+    }
+
+    return GlobalTracer.get();
+  }
+
+  public static SpanContext byteStringToSpanContext(ByteString byteString) {
+    if (byteString == null || byteString.isEmpty()) {
+      LOG.debug("The provided serialized context was null or empty");
+      return null;
+    }
+
+    SpanContext context = null;
+    ByteArrayInputStream stream =
+        new ByteArrayInputStream(byteString.toByteArray());
+
+    try {
+      ObjectInputStream objStream = new ObjectInputStream(stream);
+      Map<String, String> carrier =
+          (Map<String, String>) objStream.readObject();
+
+      context = GlobalTracer.get().extract(Format.Builtin.TEXT_MAP,
+          new TextMapExtractAdapter(carrier));
+    } catch (Exception e) {
+      LOG.warn("Could not deserialize context {}", e);
+    }
+
+    return context;
+  }
+
+  public static ByteString spanContextToByteString(SpanContext context) {
+    if (context == null) {
+      LOG.debug("No SpanContext was provided");
+      return null;
+    }
+
+    Map<String, String> carrier = new HashMap<String, String>();
+    GlobalTracer.get().inject(context, Format.Builtin.TEXT_MAP,
+        new TextMapInjectAdapter(carrier));
+    if (carrier.isEmpty()) {
+      LOG.warn("SpanContext was not properly injected by the Tracer.");
+      return null;
+    }
+
+    ByteString byteString = null;
+    ByteArrayOutputStream stream = new ByteArrayOutputStream();
+
+    try {
+      ObjectOutputStream objStream = new ObjectOutputStream(stream);
```

Comment:
This is dangerous
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/ObjectInputStream.html
"Warning: Deserialization of untrusted data is inherently dangerous and should be avoided."
We need an alternative.

Reply:
Moreover, ObjectInputStream doesn't guarantee compatibility between a server and a client if they are on different version. Finally, I bet it is not optimal serialization format. We should find one that's as short as possible.

Reply:
Good find @jojochuang . I agree we should definitely use a more optimized solution for serialization/deserialization if possible.